recipes/cuda-cupti/meta.yaml

```diff
@@ -0,0 +1,177 @@
+    - test -L $PREFIX/lib/libcupti.so.{{ lib_version }}                                      # [linux]
+    - test -L $PREFIX/targets/{{ target_name }}/lib/libcupti.so.{{ version.split(".")[0] }}  # [linux]
+    - test -f $PREFIX/targets/{{ target_name }}/lib/libcupti.so.{{ lib_version }}            # [linux]
+    - if not exist %LIBRARY_LIB%\cupti64_{{ lib_version }}.dll exit 1              # [win]
+
+outputs:
+  - name: cuda-cupti
+    files:
+      - lib/libcupti*.so.*                    # [linux]
+      - targets/{{ target_name }}/lib/*.so.*  # [linux]
+      - Library\lib\*.dll                     # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+    # Tests are defined at the top level, due to package/output name conflicts.
+    about:
+      home: https://developer.nvidia.com/cuda-toolkit
+      license_file: LICENSE
+      license: LicenseRef-NVIDIA-End-User-License-Agreement
+      license_url: https://docs.nvidia.com/cuda/eula/index.html
+      summary: Provides libraries to enable third party tools using GPU profiling APIs.
+      description: |
+        Provides libraries to enable third party tools using GPU profiling APIs.
+      doc_url: https://docs.nvidia.com/cuda/cupti/index.html
+
+  - name: cuda-cupti-dev
+    files:
+      - lib/checkpoint.so                         # [linux]
+      - lib/libcupti.so                           # [linux]
+      - lib/libnv*.so                             # [linux]
+      - lib/libpcsamplingutil.so                  # [linux]
+      - targets/{{ target_name }}/lib/*.so        # [linux]
+      - targets/{{ target_name }}/include         # [linux]
+      - Library\include                           # [win]
+      - Library\lib\*.lib                         # [win]
+    requirements:
+      build:
+        - {{ compiler("c") }}
+        - {{ compiler("cxx") }}
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+      host:
+        - cuda-version {{ cuda_version }}
+      run:
+        - {{ pin_compatible("cuda-version", max_pin="x.x") }}
+        - {{ pin_subpackage("cuda-cupti", exact=True) }}
+      run_constrained:
+        - arm-variant * {{ arm_variant_type }}  # [aarch64]
+        - cuda-cupti-static >={{ version }}
+    test:
+      commands:
+        - test -L $PREFIX/lib/libcupti.so                                    # [linux]
+        - test -L $PREFIX/lib/libnvperf_host.so                              # [linux]
+        - test -L $PREFIX/lib/libnvperf_target.so                            # [linux]
+        - test -f $PREFIX/targets/{{ target_name }}/include/cupti.h          # [linux]
```

Comment:
Should the header exist in `$PREFIX/include` as well via a symlink?

Reply:
See comment here: https://github.com/conda-forge/staged-recipes/pull/22127#discussion_r1168746777